include/mxnet/engine.h

```diff
@@ -74,15 +74,15 @@ class CallbackOnComplete {
  public:
   // use implicit copy and assign
   /*! \brief involve the callback */
-  inline void operator()(const dmlc::Error* error = nullptr) const {
+  inline void operator()(const std::exception* error = nullptr) const {
```

Comment:
Is this to support horovod to handle non dmlc exceptions ?

Reply:
No, there is no need for this in Horovod at this moment. dmlc exception is a subclass of std::exception. I feel it is better to be more generic on the interface. Additionally, your recent improvements on exception handling also catches a broader range to exceptions using std::exception instead of dmlc::Error.

Reply:
okay sure.

Reply:
Is it possible to add a unit test? For example, purposely performing  `nd.ones((1,) + extremely_small_number` to cause `std::stof ` to fail?

Reply:
I changed [this place](https://github.com/apache/incubator-mxnet/pull/14911/commits/ee563d0edaf12e0d62112680e8a0bbb1a18501e4#diff-a3398a203d3c98a27e506bf3215afa18R385) to use this new interface. A lot existing negative test cases have already covered it.

Reply:
The link was broken. Do you mind sending the link again?